src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
> The wrapX setting will be ignored for projections that return false for isGlobal().

This is what I'm questioning.  I may be interested to use OpenLayers to repeat a world that is not the earth (i.e. that is not considered "global" by OpenLayers).